### ISSUE TITLE
Unfocused "Application has crashed"

### ISSUE THREAD

**Issue by teamcons** [elementary/gala#2078]
### What Happened?

Whenever an application is unresponsible, the "Application has crashed"

very often while trying to get a response out of the app, so the dialog gets focused out immediately, leaving a frozen window that does nothing

### Steps to Reproduce

have an app that crashes, click literally anywhere else

Typing a password after confirming a list of actions on a device on Gnome Disks cause an consecutive endless stream of crash dialogs, you can try there

### Expected Behavior

Crash dialog stays on top of frozen app no matter what

### OS Version

8.x (Early Access)

### Session Type

Secure Session (Wayland)

### Software Version

Latest release (I have run all updates)

**Comment by leolost2605** [comment]
I'm don't quite understand what you mean and I can't reproduce anything similar. Could you maybe provide a screen recording? 

Also are you running elementary OS or did you install the desktop on some other distro e.g. nix or arch?

**Comment by teamcons** [comment]
> I don't quite understand what you mean and I can't reproduce anything similar. Could you maybe provide a screen recording?
> 
> Also are you running elementary OS or did you install the desktop on some other distro e.g. nix or arch?

eOS 8,
the dialog that appears when an application is frozen slips behind the application window
it should be in front of the frozen app window

i do not know how i can formulate it differently.

**Comment by leolost2605** [comment]
Ah alright it sounded like there was something else going on :)

Just noting for reference: this is probably an elementary portals bug but I don't wanna transfer yet because I'm not 100% sure

**Comment by Marukesu** [comment]
On wayland, gala doesn't send a window handle to the portal. So, it cannot set the correct parenting relation on the dialog.

I believe this would need that gala has the ability to set the parenting relation itself.

**Comment by leolost2605** [comment]
Yeah but on wayland the dialog should stay above everything if it doesn't get a parent :shrug:
